src/components/navBar/navBar.js

```diff
@@ -0,0 +1,400 @@
+
+  // State variables.
+
+  /** @private {?Element} */
+  ctrl.domEl_;
+
+  /** @private {?suiteTypes.jQuery} */
+  ctrl.inkbar_;
+
+  // need to wait for transcluded content to be available
+  var deregisterTabWatch = ctrl.scope_.$watch(
+      function() {
+        return ctrl.element_.querySelectorAll('.md-nav-button').length;
+      },
+      function(newLength) {
+        if (newLength > 0) {
+          initTabs_();
+          deregisterTabWatch();
+        }
+      });
+
+  /**
+   * Initializes the tab components once they exist.
+   * @private
+   */
+  function initTabs_() {
+    ctrl.inkbar_ = angular.element(ctrl.element_.querySelector('.md-ink-bar'));
+    ctrl.domEl_ = ctrl.element_.querySelector('.md-nav-bar');
+
+    ctrl.timeout_(function() { updateTabs_(ctrl.selectedTabName, undefined); });
+
+    ctrl.scope_.$watch('ctrl.selectedTabName', function(newValue, oldValue) {
+      // Wait a digest before update tabs for products doing
+      // anything dynamic in the template.
+      ctrl.timeout_(updateTabs_.bind(ctrl, newValue, oldValue));
+    });
+  }
+
+  /**
+   * Set the current tab to be selected.
+   * @param {string|undefined} newValue New current tab name.
+   * @param {string|undefined} oldValue Previous tab name.
+   * @private
+   */
+  function updateTabs_(newValue, oldValue) {
+    var tabs = getTabs_();
+
+    var oldIndex;
+    if (oldValue) {
+      var oldTab = getTabByName_(oldValue);
+      if (oldTab) {
+        oldTab.setSelected(false);
+        oldIndex = tabs.indexOf(oldTab);
+      }
+    }
+
+    if (newValue) {
+      var tab = getTabByName_(newValue);
+      if (tab) {
+        tab.setSelected(true);
+        var newIndex = tabs.indexOf(tab);
+        ctrl.timeout_(function() {
+          updateInkBarStyles(tab, newIndex, oldIndex);
+        });
+      }
+    }
+  }
+
+  /**
+   * Repositions the ink bar to the selected tab.
+   * "borrowed" from ng-material
+   * @export
+   */
+  function updateInkBarStyles(tab, newIndex, oldIndex) {
+    var tabEl = tab.getButtonEl();
+    var left = tabEl.offsetLeft;
+
+    ctrl.inkbar_.toggleClass('md-left', newIndex < oldIndex)
+        .toggleClass('md-right', newIndex > oldIndex);
+    ctrl.inkbar_.css({left: left + 'px', width: tabEl.offsetWidth + 'px'});
+  }
+
+  /**
+   * Returns an array of the current tabs.
+   * @return {!Array<!NavLinkController>}
+   * @private
+   */
+  function getTabs_() {
+    var linkArray = Array.prototype.slice.call(
+        ctrl.element_.querySelectorAll('.md-nav-link'));
+    return linkArray.map(function(el) {
+      return angular.element(el).controller('mdNavLink')
+    });
+  }
+
+  /**
+   * Returns the tab with the specified name.
+   * @param {string} name The name of the tab, found in its data-name attribute.
+   * @return {!NavLinkController|undefined}
+   * @private
+   */
+  function getTabByName_(name) {
+    return findTab_(function(tab) { return tab.getName() == name; });
```

Comment:
optional: personally I'm not a fan of the single-lined functions, but this project doesn't officially have a rule against it.